api/src/main/java/io/druid/data/input/impl/prefetch/RetryingInputStream.java

```diff
@@ -67,27 +67,30 @@ private boolean isConnectionReset(Throwable t)
            (t.getCause() != null && isConnectionReset(t.getCause()));
   }
 
-  private void waitOrThrow(Throwable t, int i) throws IOException
+  private void waitOrThrow(Throwable t, int nTry) throws IOException
   {
     final boolean isConnectionReset = isConnectionReset(t);
     if (isConnectionReset || retryCondition.apply(t)) {
       if (isConnectionReset) {
         // Re-open the input stream on connection reset
         try {
           startOffset += delegate.getCount();
```

Comment:
Not sure exception in getCount() should be ignored.

Reply:
I can't imagine what kinds of exceptions can occur except NPE. ```delegate``` shouldn't be null here, so NPE is also not the case.

Reply:
I thought that getCount() throws IOException, if it was inside try-catch. Please move it out of try-catch for clarity.